tests/containers/podman_remote.pm

```diff
@@ -0,0 +1,75 @@
+
+    select_serial_terminal();
+    my $podman = $self->containers_factory('podman');
+    $self->{podman} = $podman;
+
+    # Prepare ssh from root to the normal user
+    systemctl 'enable --now sshd';
+    assert_script_run("loginctl enable-linger $testapi::username");
+    assert_script_run('ssh-keygen -t ecdsa -N "" -C "podman-remote" -f ~/.ssh/podman-remote_ecdsa');
+    assert_script_run "install -o $testapi::username -g users -m 0700 -d /home/$testapi::username/.ssh";
+    assert_script_run "install -o $testapi::username -g users -m 0644 ~/.ssh/podman-remote_ecdsa.pub /home/$testapi::username/.ssh/authorized_keys";
+
+    # Pull and run a container named test-root
+    assert_script_run "podman pull $image";
+    assert_script_run "podman run -d --rm --name test-root $image sleep infinity";
+
+    # Run the podman daemon as normal user
+    select_user_serial_terminal();
+    assert_script_run('podman --remote help');
+    systemctl '--user enable --now podman.socket';
+
+    # Pull and run a container named test-user
+    assert_script_run "podman pull $image";
+    assert_script_run "podman run -d --rm --name test-user $image sleep infinity";
+
+    # Add the podman-remote connection to the user's podman via SSH
+    select_serial_terminal();
+    my $uid = script_output "id -u $testapi::username";
+    my $connection = "ssh://$testapi::username@127.0.0.1:22/run/user/$uid/podman/podman.sock";
+    assert_script_run "podman --remote system connection add test --identity ~/.ssh/podman-remote_ecdsa $connection";
+    validate_script_output 'podman --remote system connection list', sub { m@$connection@; };
+
+    # Validate podman-remote sees containers launched by the user (as it is connected to user's podman daemon) and not containers launched by root
+    validate_script_output 'podman --remote container ls', sub { m/test-user/; }, fail_message => "The podman-remote is connected to user's session so it should see see containers launched by user.";
+    validate_script_output 'podman --remote container ls', sub { !m/test-root/; }, fail_message => "The podman-remote is connected to user's session so it should not see containers launched by root.";
+
+    # Validate podman sees containers launched by root (as it is executed under root) and not containers launched by user
+    validate_script_output 'podman container ls', sub { m/test-root/; }, fail_message => "The podman is running under root so it should see containers initiated by root.";
+    validate_script_output 'podman container ls', sub { !m/test-user/; }, fail_message => "The podman is running under root so it should not see containers initiated by user.";
+}
+
+sub cleanup {
+    my ($self) = @_;
+    select_serial_terminal();
+    $self->{podman}->cleanup_system_host();
+}
+
+sub post_run_hook {
+    my ($self) = @_;
+    $self->cleanup();
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    $self->cleanup();
+}
+
```

Comment:
Using systemd lingers might be invasive to the system, so I think here it might be good to add a snapshot and restore the test afterwards:

```

sub test_flags {
    return {always_rollback => 1};
}
```

Reply:
Better would be to include the lingering check in cleanup, test for `/var/lib/systemd/linger/$testapi::username` and just run the disable command

Reply:
but since we are not closing the user's session, is even lingering necessary?

Reply:
True, thank you Martin, thank you Felix.